libraries/profile.rb

```diff
@@ -141,6 +141,22 @@ class ComplianceProfile < Chef::Resource
         end
       end
 
+      action :upload do
+        converge_by 'upload compliance profile' do
+          raise 'unable to determine path to compressed profile archive' if path.nil?
+          o, p = normalize_owner_profile
+          node.run_state['compliance'] ||= {}
+
+          if node.run_state['compliance']['access_token']
+            reqpath ="owners/#{o}/compliance/#{p}/tar"
+            url = construct_url(server, reqpath)
+            Chef::Log.info "Upload compliance profile #{o}/#{p} to: #{url}"
+            res = Compliance::HTTP.post_file(url.to_s, node.run_state['compliance']['access_token'], path, insecure)
```

Comment:
Currently that is not using the upload function for inspec https://github.com/chef/inspec/blob/master/lib/bundles/inspec-compliance/api.rb#L82 Any specific reason?

We need to be aware that the current method is not using the checking mechanisms as available. https://github.com/chef/inspec/blob/master/lib/bundles/inspec-compliance/cli.rb#L94-L167

Do you think we could combine them? I would make to just reuse functions we already have. If that requires refactoring in InSpec, I would go for that.